components/ILIAS/Forum/classes/Moderation/class.ForumModeratorsTable.php

```diff
@@ -0,0 +1,216 @@
+    {
+        $columns = $this->getColumns();
+        $actions = $this->getActions();
+        $data_retrieval = $this->getDataRetrieval();
+
+        $table = $this->ui_factory->table()
+                                  ->data($this->lng->txt('frm_moderators'), $columns, $data_retrieval)
+                                  ->withActions($actions)
+                                  ->withRequest($this->request);
+
+        return $table;
+    }
+
+    protected function getColumns(): array
+    {
+        $columns = [
+            'usr_id' => $this->ui_factory->table()->column()->number('User ID')
+                                         ->withIsSortable(false),
+
+            'login' => $this->ui_factory->table()->column()->text($this->lng->txt('login'))
+                                        ->withIsSortable(true),
+
+            'firstname' => $this->ui_factory->table()->column()->text($this->lng->txt('firstname'))
+                                            ->withIsSortable(true),
+
+            'lastname' => $this->ui_factory->table()->column()->text($this->lng->txt('lastname'))
+                                           ->withIsSortable(true),
+        ];
+
+        return $columns;
+    }
+
+    protected function getActions(): array
+    {
+        $query_params_namespace = ['frm_moderators_table'];
+
+        $uri_detach = $this->data_factory->uri(
+            ILIAS_HTTP_PATH . '/' . $this->ctrl->getLinkTargetByClass('ilforummoderatorsgui', 'detachModeratorRole')
+        );
+
+        $url_builder_detach = new UI\URLBuilder($uri_detach);
+        list(
+            $url_builder_detach, $action_parameter_token_copy, $row_id_token_detach
+            ) =
+            $url_builder_detach->acquireParameters(
+                $query_params_namespace,
+                'action',
+                'usr_ids'
+            );
+
+        $actions = [
+            'detachModeratorRole' => $this->ui_factory->table()->action()->multi(
+                $this->lng->txt('remove'),
+                $url_builder_detach->withParameter($action_parameter_token_copy, 'detachModeratorRole'),
+                $row_id_token_detach
+            ),
+        ];
+        return $actions;
+    }
+
+    protected function getDataRetrieval(): UI\Component\Table\DataRetrieval
+    {
+        $data_retrieval = new class($this->forum_moderators) implements UI\Component\Table\DataRetrieval {
+
+            public function __construct(
+                protected \ilForumModerators $forum_moderators
+            ) {
+            }
+
+            public function getRows(
+                UI\Component\Table\DataRowBuilder $row_builder,
+                array $visible_column_ids,
+                Data\Range $range,
+                Data\Order $order,
+                ?array $filter_data,
+                ?array $additional_parameters
+            ): \Generator {
+                $records = $this->getRecords($range);
```

Comment:
We should pass the `$order` as well and sort the array accordingly. I noticed you added a `orderRecords` method below which seems to be not in use.

Reply:
done. Thanks for review.